### ISSUE TITLE
Engine signature invalid pointer

### ISSUE THREAD

**Issue by iekaslo** [roflmuffin/CounterStrikeSharp#360]
Hello.
Signatures that are requested from the engine library cannot be found and used without explicitly specifying binaryPath where the absolute path to the libengine2 library will be written.

```
"signatures": {
	  "library": "engine",
          "linux": "\\x55\\x22..."
          "windows": "\\x55\\x22..."
}
```

**Comment by KillStr3aK** [comment]
yeah, I've only adjusted the code to allow passing arbitrary binaries to the internal managed functions but did not touched gamedata related parts, should be an easy fix

**Comment by KillStr3aK** [comment]
> yeah, I've only adjusted the code to allow passing arbitrary binaries to the internal managed functions but did not touched gamedata related parts, should be an easy fix

well turns out it's not that easy
I mean its not a big deal in general, we just have to consider how we return the gamedata values, a change like this would require a new type that returns the whole data, not just the signature as we can't guess what binary the signature is in, but having more data could erase this issue

currently when the binarypath is not set explicitly we always assume `server`

https://github.com/roflmuffin/CounterStrikeSharp/blob/42dd270b7852c7d866a8bdf835524fc0bd27a787/managed/CounterStrikeSharp.API/Modules/Memory/DynamicFunctions/BaseMemoryFunction.cs#L19
